internal/router/router.go

```diff
@@ -476,14 +476,17 @@ func (r *Router) serveBulkMetaGet(req *http.Request) (interface{}, error) {
 	// TODO get the metadata concurrently for each id.
 	ids := req.Form["id"]
 	if len(ids) == 0 {
-		return nil, errgo.Newf("no ids specified in meta request")
+		return nil, errgo.WithCausef(nil, params.ErrBadRequest, "no ids specified in meta request")
 	}
 	delete(req.Form, "id")
 	result := make(map[string]interface{})
 	for _, id := range ids {
 		url, err := charm.ParseReference(id)
 		if err != nil {
-			return nil, errgo.Mask(err)
+			// Failing to parse a reference should be treated as if the charm cannot be
```

Comment:
I'm not convinced about this. For charm ids in the URL path that's true, but I think people should be providing valid charm URLs to the id attribute.